### DISCUSSION TITLE
Just curious - how does Github decide a language's color?

### DISCUSSION THREAD

**Discussion by zarahzachz** [github-linguist/linguist#6560] [General]
I found a list of all the hexcode colors for each programming and markup language and was curious how they got the colors they were assigned.

**Comment by lildude** [comment]
It depends 😁 

Some of the oldest languages were chosen at random or loosely based on logos or branding at the time many many many years ago. At the time there was a requirement that no two languages could match. Later a requirement was introduced that no two languages could have colours that are visually similar to each other, so some were changed then with little reasoning other than to ensure they're visually distinct. 

The visually distinct requirement has now been dropped so we only disallow identical colours now.

From that point onwards we've asked for substantiation for a colour when a new language is added, and recently added it to the PR template so this can easily be found later when someone asks.

Before anyone gets overly excited and decides to submit a PR to change a whole bunch of colours, we have a hard requirement that any language colour change _must_ get community support on that language's public community forums, chat or whatever etc, and it must be referenced in the PR template. This requirement was introduced after there was major fallout when the Rust language was changed in https://github.com/github-linguist/linguist/pull/4319 .

**Reply by zarahzachz** [reply]
lol typical Rust 😂

Thanks for responding to my silly question
